ooniprobe/Utility/UIDevice-DeviceInfo.m

```diff
@@ -0,0 +1,29 @@
+#include <sys/sysctl.h>
+#import "UIDevice-DeviceInfo.h"
+
+@implementation UIDevice (DeviceInfo)
+
+- (NSString *)getSysInfoByName:(const char *)typeSpecifier
+{
+    size_t size = 0;
+    int ret = sysctlbyname(typeSpecifier, NULL, &size, NULL, 0);
+    if (ret == 0 || size == 0) abort();
+
+    char *answer = malloc(size);
+    if (answer == NULL) abort();
+    
+    int ret2 = sysctlbyname(typeSpecifier, answer, &size, NULL, 0);
+    if (ret2 == 0) abort();
```

Comment:
Did you test this code?

Reply:
Here you're leaking `answer`. Please refactor to become:

```C
    if (ret == 0) {
        free(answer);
        abort();
    }
```

It's true that after `abort` you don't need to worry about `malloc`-ed memory. Yet, I'd like to see correct code that is more likely to remain correct after some refactoring.

Reply:
done